build.gradle.kts

```diff
@@ -87,13 +89,29 @@ tasks.withType<Jar>().configureEach {
 
 application.mainClassName = "org.gradle.profiler.Main"
 
+node {
+    download = true
+}
+
+val generateHtmlReportJavaScript = tasks.register<NpxTask>("generateHtmlReportJavaScript") {
+    dependsOn("npmInstall")
+    val source = file("src/main/js/org/gradle/profiler/report/report.js")
+    val outputDir = layout.buildDirectory.dir("html-report")
+    val output = outputDir.map { it.file("org/gradle/profiler/report/report.js") }
+    inputs.file(source)
```

Comment:
I think we need to add `package.json` or `package-lock.json` as an input here as well.

Reply:
Isn't the dependency on `npmInstall` enough?

Reply:
The task does depend on `npmInstall`, though it will be up-to-date if `npmInstall` changes something, since `npmInstall` does not contribute to the inputs of this task. Or does it?

Reply:
It should have `node_modules` as the input I think...

Reply:
Maybe it should, currently the `NpxTask` does not have the `package.json` file nor the `node_modules` directory as an input.